server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -122,7 +129,8 @@ public void start(
         MetaStateService metaStateService,
         MetadataIndexUpgradeService metadataIndexUpgradeService,
         MetadataUpgrader metadataUpgrader,
-        PersistedClusterStateService persistedClusterStateService
+        PersistedClusterStateService persistedClusterStateService,
+        RemoteClusterStateService remoteClusterStateService
```

Comment:
With the current integration, it will always initialize RemoteClusterStateService even when remote cluster state is not enabled.

Reply:
RemoteClusterStateService is not a heavy object. It is contains few fields (<10). Anyway I will see if can initialize this only when remote state is enabled.